src/operator/tensor/indexing_op.h

```diff
@@ -341,6 +341,53 @@ struct Take {
   }
 };
 
+template<bool clip = true>
```

Comment:
Why are you moving device-specific kernel to `.h` ?

Reply:
Are you looking to share the kernels with the numpy version?

Reply:
> Are you looking to share the kernels with the numpy version?

Yes, I want to reuse kernels of ndarray version.